### PR TITLE
fix(blockstore): blocksync fixes ported from 0.38.8

### PR DESCRIPTION
**Pull request by greg-szabo**
Fixes [ASA-2024-008](https://github.com/cometbft/cometbft/security/advisories/GHSA-hg58-rf2h-6rr7) on the v1.x branch.

---

#### PR checklist

- [X] Tests written/updated
- [X] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [X] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Comment by greg-szabo** [comment]
The decision made [here](https://github.com/tendermint/tendermint/pull/8433#discussion_r866790631) to panic when no extended commit is found was not ported back to v0.38 and/or v0.37. Because of that, TestCheckExtendedCommitMissing fails on main and v1.x while it succeeds on v0.38.8 and v0.37.7 with the fixes in ASA-2024-008. (This PR.)

I'm planning to catch the panic in the test on main and v1.x and fail if the test did not panic. Currently, the issue is that the panic happens in a sub-thread that testify's `require.Panics` does not seem to catch. I'm open for ideas.

File: .changelog/unreleased/bug-fixes/0000-asa-2024-008.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- `[blockstore]` Added peer banning in blockstore
```
Comment (melekes) [inline .changelog/unreleased/bug-fixes/0000-asa-2024-008.md:1]:
```suggestion
- `[blocksync]` Added peer banning
```

**Review by melekes — APPROVED** [review]
👍

**Review by sergio-mena — COMMENTED** [review]
LGTM. Thanks for the extra effort!

**Pull request merged**